features/step_definitions/plans/application_plans_steps.rb

```diff
@@ -131,3 +131,27 @@
   assert_content "The plan was deleted", wait: 10
   step(%(I should not see plan "#{name}"))
 end
+
+And "an application plan that is not default" do
+  @plan = FactoryBot.create(:application_plan, name: "Magic", issuer: @provider.default_service)
+  assert_not @plan.master?
+end
+
+Given "the application plan is hidden" do
+  assert @plan.hidden?
+end
+
+Then "an admin can select the application plan as default" do
+  select_default_plan(@plan.name)
+  assert has_content? "The default plan has been changed."
+  assert @plan.reload.master?
+end
+
+Given "the application plan has been deleted" do
+  @plan.destroy
+end
+
+Then "an admin can't select the application plan as default" do
+  select_default_plan(@plan.name)
+  assert has_content? "Not Found"
```

Comment:
this is a little weird, we try to click on a plan that was deleted in the background? Do we need such test?

Reply:
For some reason it was there. I think it's possible a user deletes the plan while other is selecting is as default at the same time. Rare but possible 😄